plugin-server/tests/main/teardown.test.ts

```diff
@@ -48,16 +62,30 @@ describe('teardown', () => {
             undefined
         )
 
-        const error1 = await getErrorForPluginConfig(pluginConfig39.id)
-        expect(error1).toBe(null)
-
         await processEvent(hub!, defaultEvent)
 
-        await stop?.()
+        await stop!()
 
-        // verify the teardownPlugin code runs
-        const error2 = await getErrorForPluginConfig(pluginConfig39.id)
-        expect(error2.message).toBe('This Happened In The Teardown Palace')
+        // verify the teardownPlugin code runs -- since we're reading from
+        // ClickHouse, we need to give it a bit of time to have consumed from
+        // the topic and written everything we're looking for to the table
+        await waitForExpect(async () => {
+            const logEntries = await getLogEntriesForPluginConfig(hub!, pluginConfig39.id)
+
+            const systemErrors = logEntries.filter(
+                (logEntry) =>
+                    logEntry.source == PluginLogEntrySource.System && logEntry.type == PluginLogEntryType.Error
+            )
+            expect(systemErrors).toHaveLength(1)
+            expect(systemErrors[0].message).toContain('Plugin failed to unload')
+
+            const pluginErrors = logEntries.filter(
+                (logEntry) =>
+                    logEntry.source == PluginLogEntrySource.Plugin && logEntry.type == PluginLogEntryType.Error
+            )
+            expect(pluginErrors).toHaveLength(1)
+            expect(pluginErrors[0].message).toContain('This Happened In The Teardown Palace')
+        })
```

Comment:
This test (and the one below) was kind of weird before in how it uses the error state of the plugin to check that some code was/was not run, but it's even more weird now that it relies on the whole Kafka and ClickHouse plugin log pipeline. There's surely some more reasonable approach here, but I didn't take the time to look into what that would be.

Reply:
This is also going to be impacted by #18816, probably worth figuring out a less brittle approach at this point.